agentic-tutorial/src/main/java/_5_conditional_workflow/_5b_Conditional_Workflow_Example_Async.java

```diff
@@ -0,0 +1,99 @@
+package _5_conditional_workflow;
+
+import _4_parallel_workflow.ManagerCvReviewer;
+import dev.langchain4j.agentic.AgenticServices;
+import dev.langchain4j.agentic.UntypedAgent;
+import dev.langchain4j.model.chat.ChatModel;
+import domain.CvReview;
+import util.ChatModelProvider;
+import util.StringLoader;
+import util.log.CustomLogging;
+import util.log.LogLevels;
+
+import java.io.IOException;
+import java.util.Map;
+
+public class _5b_Conditional_Workflow_Example_Async {
+
+    static {
+        CustomLogging.setLevel(LogLevels.PRETTY, 150);
+    }
+
+    /**
+     * This example demonstrates multiple fulfilled conditions and async agents that will
+     * allow consecutive agents to be called in parallel for faster execution.
+     * In this example:
+     * - condition 1: if the HrReview is good, the CV is passed to the manager for review,
+     * - condition 2: if the HrReview indicates missing information, the candidate is contacted for more info.
+     */
+
+    private static final ChatModel CHAT_MODEL = ChatModelProvider.createChatModel();
+
+    public static void main(String[] args) throws IOException {
+
+        // 1. Create all async agents
+        ManagerCvReviewer managerCvReviewer = AgenticServices.agentBuilder(ManagerCvReviewer.class)
+                .chatModel(CHAT_MODEL)
+                .async(true) // async agent
+                .outputName("managerReview")
+                .build();
+        EmailAssistant emailAssistant = AgenticServices.agentBuilder(EmailAssistant.class)
+                .chatModel(CHAT_MODEL)
+                .async(true)
+                .tools(new OrganizingTools())
+                .outputName("sentEmailId")
+                .build();
+        InfoRequester infoRequester = AgenticServices.agentBuilder(InfoRequester.class)
+                .chatModel(CHAT_MODEL)
+                .async(true)
+                .tools(new OrganizingTools())
+                .outputName("sentEmailId")
+                .build();
+
+        // 2. Build async conditional workflow
+        UntypedAgent candidateResponder = AgenticServices
+                .conditionalBuilder()
+                .subAgents(scope -> {
+                    CvReview hrReview = (CvReview) scope.readState("cvReview");
+                    return hrReview.score >= 0.8; // if HR passes, send to manager for review
+                }, managerCvReviewer)
+                .subAgents(scope -> {
+                    CvReview hrReview = (CvReview) scope.readState("cvReview");
+                    return hrReview.score < 0.8; // if HR does not pass, send rejection email
+                }, emailAssistant)
+                .subAgents(scope -> {
+                    CvReview hrReview = (CvReview) scope.readState("cvReview");
+                    return hrReview.feedback.toLowerCase().contains("missing information:");
+                }, infoRequester) // if needed, request more info from candidate
+                .output(agenticScope ->
```

Comment:
With [this fix](https://github.com/langchain4j/langchain4j/commit/9443ad9b2e0726a37611a026e2dea049bb1f7097) (that however will be available with 1.6.0 version) having an output function should be no longer necessary. That said it seems a bit odd to me that you have some agentic invocation not producing any meaningful output, but only relying on the agents' side effects. Maybe this is reasonable in some situation like this, but I'm not sure if this is a usage pattern that we want to promote.